lib/functions.php

```diff
@@ -214,16 +214,17 @@ function wait($promise)
         $resolved = false;
 
         try {
-            do {
-                Loop::run(function () use (&$resolved, &$value, &$exception, $promise) {
-                    $promise->onResolve(function ($e, $v) use (&$resolved, &$value, &$exception) {
-                        Loop::stop();
-                        $resolved = true;
-                        $exception = $e;
-                        $value = $v;
-                    });
-                });
-            } while (!$resolved && Loop::getInfo()['enabled_watchers']['referenced'] > 0);
+            $promise->onResolve(function ($e, $v) use (&$resolved, &$value, &$exception) {
+                Loop::stop();
+
+                $resolved = true;
+                $exception = $e;
+                $value = $v;
+            });
+
+            while (!$resolved && Loop::getInfo()['enabled_watchers']['referenced'] > 0) {
```

Comment:
Should we make `Driver::isEmpty()` public so this doesn't feel as hacky?

Reply:
This would be a slight breaking change for drivers like `TracingDriver`, while the current implementation uses public API.